src/components/ui/Chat/ChannelList/ChannelItem.tsx

```diff
@@ -22,25 +22,72 @@ export interface ChannelItemProps
   isSelected?: boolean;
   /** Callback function when clicked */
   onClick: () => void;
-  /** Signifies of there are unread message in the channel. */
+  /** Signifies if there are unread message in the channel. */
   unread?: boolean;
   /** Content of the badge signaling unread messages, such as a badge displaying the number. */
   unreadBadgeLabel?: string;
+  /* Content of the last channel message to be displayed */
+  lastChannelMessage?: string;
+  /* Timestamp of the last channel message to be displayed */
+  lastChannelMessageTimestamp?: string;
 }
 
 export const ChannelItem: FC<ChannelItemProps> = (props) => {
-  const { name, actions, isSelected, onClick, unread, unreadBadgeLabel } =
-    props;
+  const {
+    name,
+    actions,
+    isSelected,
+    onClick,
+    unread,
+    unreadBadgeLabel,
+    lastChannelMessage,
+    lastChannelMessageTimestamp,
+  } = props;
+
+  const displayDetailedView = lastChannelMessage || lastChannelMessageTimestamp;
+  const displayUnreadBatch = unread && unreadBadgeLabel;
+  const displayPopOver = actions && isSelected;
   return (
     <StyledChannelItem
       {...props}
       className={classnames({ 'ch-selected': isSelected, 'ch-unread': unread })}
     >
-      <Button className="ch-channel-button" label={name} onClick={onClick} />
-      {unread && unreadBadgeLabel && (
-        <Badge value={unreadBadgeLabel} className="ch-unread-badge" />
+      {displayDetailedView && (
+        <div className={'ch-detailed-channel'} onClick={onClick}>
+          <div className="ch-detailed-channel-name">{name}</div>
+          <div className="ch-detailed-channel-message">{lastChannelMessage}</div>
+          <div className="ch-detailed-channel-message-time">
+            {lastChannelMessageTimestamp}
+          </div>
+        </div>
+      )}
+      {displayDetailedView && displayUnreadBatch && (
+        <Badge value={unreadBadgeLabel!} className="ch-unread-badge-detailed" />
+      )}
+      {displayDetailedView && displayPopOver && (
```

Comment:
do you need to repeat the `displayDetailedView` for each conditional? 

Maybe its cleaner to just do it once?

Reply:
Tried to do it once for all components. Could not get it compiled, so did it this way. Please let me know if I missed anything thanks.

Reply:
```
{
    displayDetailedView && (
       // Logic inside here?
    )
}
```

Or you can use ternary operator (?) to be explicit about what to render.

```
{ 
   displayDetailedView ? (
       // Logic inside here?
   ) : <></>
}
```

What error did you get?

Reply:
But, actually, I'm fine with your approach too since it reduces another child level.

Reply:
```
displayDetailedView ? (
  <>
    <div className={'ch-detailed-channel'} onClick={onClick}>
      <div className="ch-detailed-channel-name">{name}</div>
      <div className="ch-detailed-channel-message">{lastChannelMessage}</div>
      <div className="ch-detailed-channel-message-time">
        {lastChannelMessageTimestamp}
      </div>
    </div>
    {displayUnreadBatch && (
      <Badge value={unreadBadgeLabel!} className="ch-unread-badge-detailed" />
    )}
    {displayPopOver && (
      <PopOver
        className={'ch-popover-toggle-detailed'}
        a11yLabel="Open channel options"
        placement="bottom-end"
        renderButton={(isOpen: boolean) => (
          <Dots
            width="1.5rem"
            height="1.5rem"
            className={`${isOpen ? 'isOpen' : ''} ch-channel-actions`}
            data-testid="channel-actions"
          />
        )}
        children={actions}
      />
    )}
  </>
) : (
  <>
    <Button className="ch-channel-button" label={name} onClick={onClick} />
    {displayUnreadBatch && (
      <Badge value={unreadBadgeLabel!} className="ch-unread-badge"></Badge>
    )}
    {displayPopOver && (
      <PopOver
        a11yLabel="Open channel options"
        placement="bottom-end"
        renderButton={(isOpen: boolean) => (
          <Dots
            width="1.5rem"
            height="1.5rem"
            className={`${isOpen ? 'isOpen' : ''} ch-channel-actions`}
            data-testid="channel-actions"
          />
        )}
        children={actions}
      />
    )}
  </>
)
```